### PR TITLE
Throw TrinoException when Iceberg commit fails

### PR DESCRIPTION
**Pull request by pajaks**
## Description

In case Iceberg commit fails throw TrinoException with clear error type.

## Additional context and related issues

Example stack trace:
```
  "type": "org.apache.iceberg.exceptions.ValidationException",
  "message": "Cannot commit, found new delete for replaced data file: ",
  "suppressed": [],
  "stack": [
    "org.apache.iceberg.exceptions.ValidationException.check(ValidationException.java:49)",
    "org.apache.iceberg.MergingSnapshotProducer.validateNoNewDeletesForDataFiles(MergingSnapshotProducer.java:481)",
    "org.apache.iceberg.MergingSnapshotProducer.validateNoNewDeletesForDataFiles(MergingSnapshotProducer.java:413)",
    "org.apache.iceberg.BaseRewriteFiles.validate(BaseRewriteFiles.java:140)",
    "org.apache.iceberg.SnapshotProducer.apply(SnapshotProducer.java:233)",
    "org.apache.iceberg.SnapshotProducer.lambda$commit$2(SnapshotProducer.java:384)",
    "org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:413)",
    "org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:219)",
    "org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:203)",
    "org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:196)",
    "org.apache.iceberg.SnapshotProducer.commit(SnapshotProducer.java:382)",
    "org.apache.iceberg.BaseTransaction.applyUpdates(BaseTransaction.java:509)",
    "org.apache.iceberg.BaseTransaction.lambda$commitSimpleTransaction$3(BaseTransaction.java:425)",
    "org.apache.iceberg.util.Tasks$Builder.runTaskWithRetry(Tasks.java:413)",
    "org.apache.iceberg.util.Tasks$Builder.runSingleThreaded(Tasks.java:219)",
    "org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:203)",
    "org.apache.iceberg.util.Tasks$Builder.run(Tasks.java:196)",
    "org.apache.iceberg.BaseTransaction.commitSimpleTransaction(BaseTransaction.java:423)",
    "org.apache.iceberg.BaseTransaction.commitTransaction(BaseTransaction.java:318)",
    "io.trino.plugin.iceberg.IcebergMetadata.finishOptimize(IcebergMetadata.java:1610)",
    "io.trino.plugin.iceberg.IcebergMetadata.finishTableExecute(IcebergMetadata.java:1533)",
```

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.
( ) Release notes are required. Please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -1581,7 +1581,12 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         Snapshot snapshot = requireNonNull(icebergTable.snapshot(optimizeHandle.snapshotId().get()), "snapshot is null");
         rewriteFiles.validateFromSnapshot(snapshot.snapshotId());
         commit(rewriteFiles, session);
-        transaction.commitTransaction();
+        try {
+            transaction.commitTransaction();
```
Comment (findinpath) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:1585]: There are a few other places where `transaction.commitTransaction();` in `IcebergMetadata` is being called. Do they need wrapping as well?

**Review by findinpath — APPROVED** [review]

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -1581,7 +1581,12 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
...
+        }
+        catch (RuntimeException e) {
+            throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to commit during optimize", e);
```
Comment (ebyhr) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:1588]: Thus, users can understand the failed cause without looking into the stacktrace:
```suggestion
            throw new TrinoException(ICEBERG_COMMIT_ERROR, "Failed to commit during optimize: " + firstNonNull(e.getMessage(), e), e);
```

File: plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java [hunk 3]
```diff
@@ -1609,6 +1608,16 @@ private void finishOptimize(ConnectorSession session, IcebergTableExecuteHandle
         transaction = null;
     }
 
+    private static void commitTransaction(Transaction transaction, String operation)
+    {
+        try {
+            transaction.commitTransaction();
+        }
+        catch (RuntimeException e) {
```
Comment (losipiuk) [inline plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java:1616]: I am not sure if we want to catch everything here

Comment (pajaks) [reply plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java]: Changed to `ValidationException`

**Comment by ebyhr** [comment]
Iceberg CI failure is related to this change: https://github.com/trinodb/trino/actions/runs/10387385304/job/28761464164?pr=23008

**Review by losipiuk — APPROVED** [review]

**Pull request merged**